tasks/extract-binaries-from-image/extract-binaries-from-image.yaml

```diff
@@ -41,28 +45,50 @@ spec:
             echo "Error: No valid snapshot file was provided."
             exit 1
         fi
-        IMAGE_URL=$(jq -r '.components[0].containerImage // ""' "${SNAPSHOT_SPEC_FILE}")
-        if [ -z "${IMAGE_URL}" ] ; then
-            echo "Error: Unable to get image url from snapshot."
-            exit 1
-        fi
 
         BINARIES_DIR=binaries
         BINARIES_PATH=$(workspaces.data.path)/$(params.subdirectory)/$BINARIES_DIR
         mkdir -p $BINARIES_PATH
 
-        TMP_DIR=$(mktemp -d)
-        skopeo copy docker://$IMAGE_URL dir:$TMP_DIR
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No data JSON was provided."
+            DESIRED_COMPONENTS_LIST=()
+        elif [ $(jq '."content-gateway" | has("components")' "${DATA_FILE}") = true ]; then
+            DESIRED_COMPONENTS_LIST=$(jq -r '."content-gateway".components[].name' "${DATA_FILE}")
+        else
+            DESIRED_COMPONENTS_LIST=$()
```

Comment:
Why is it `()` on line 56 but `$()` here? Assuming they actually should be the same on the two lines, you could always set it before the if and then just do one if check for if data file exists and <jq query> then ....

Reply:
Yep sorry, that was a mistake from this last refactor.  I'll fix that now.